builtin/providers/profitbricks/data_source_datacenter.go

```diff
@@ -0,0 +1,76 @@
+package profitbricks
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/profitbricks/profitbricks-sdk-go"
+	"log"
+	"strings"
+)
+
+func dataSourceDataCenter() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDataCenterRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceDataCenterRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	profitbricks.SetAuth(config.Username, config.Password)
+
+	datacenters := profitbricks.ListDatacenters()
+
+	if datacenters.StatusCode > 299 {
+		return fmt.Errorf("An error occured while fetching datacenters %s", datacenters.Response)
+	}
+
+	name, _ := d.GetOk("name")
```

Comment:
as it's a required field we can guarantee it's there so we can write

`name := d.Get("name").(string)`